test-case/verify-pcm-list.sh

```diff
@@ -23,50 +23,31 @@ OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
 func_opt_parse_option $*
 tplg=${OPT_VALUE_lst['t']}
 
-[[ ! "$tplg" ]] && dlogw "Missing tplg file for this case" && exit 1
+tplg_path=`func_lib_get_tplg_path "$tplg"`
+[[ "$?" != "0" ]] && dloge "No available topology for this test case" && exit 1
 
 # hijack DMESG_LOG_START_LINE which refer dump kernel log in exit function
 DMESG_LOG_START_LINE=$(sof-get-kernel-line.sh|tail -n 1 |awk '{print $1;}')
 
-# check TPLG by the loop
-# TODO: current miss multiple tplg behaivor
-# so this case logic just for 1 tplg process
-while [ ${#tplg} -gt 0 ]
-do
-    # go through each TPLG file and check the PCM list
-    # left ',' 1st field
-    tplg_file=${tplg%%,*}
-    # expect left ',' 1st field
-    tplg=${tplg#*,}
-    [ "$tplg_file" == "$tplg" ] && tplg=""
-    if [ -f "$TPLG_ROOT/$(basename $tplg_file)" ]; then
-        tplg_file="$TPLG_ROOT/$(basename $tplg_file)"   # catch from TPLG_ROOT
-    elif [ -f "$tplg_file" ];then
-        tplg_file=$(realpath $tplg_file)    # relative path -> absolute path
-    else
-        dloge "Couldn't find target TPLG file $tplg_file needed to run ${BASH_SOURCE[0]}" && exit 1
-    fi
-
-    tplg_str="$(sof-tplgreader.py $tplg_file -d id pcm type -o)"
-    pcm_str="$(sof-dump-status.py -i ${SOFCARD:-0})"
-
-    dlogi "CMD: 'sof-tplgreader.py $tplg_file -d id pcm type -o' to get tplg list string:"
-    echo "$tplg_str"
-    dlogi "CMD: 'sof-dump-status.py -i ${SOFCARD:-0}' to get pcm list string:"
-    echo "$pcm_str"
-
-    if [[ "$tplg_str" != "$pcm_str" ]]; then
-        dloge "TPLG mismatch with PCM"
-        dlogi "Dump aplay -l"
-        aplay -l
-        dlogi "Dump arecord -l"
-        arecord -l
-        sof-kernel-dump.sh > $LOOG_ROOT/kernel.txt
-        exit 1
-    fi
-
-    # TODO: Miss multiple TPLG mapping pcm logic
-    break
-done
-
+tplg_str="$(sof-tplgreader.py $tplg_path -d id pcm type -o)"
+pcm_str="$(sof-dump-status.py -i ${SOFCARD:-0})"
+
+dlogc 'sof-tplgreader.py $tplg_file -d id pcm type -o'
```

Comment:
double quote. And the same to others.

Reply:
@libinyang There are a lot of $var not quoted, let's not do it partially now, and  I will do an overall cleanup on this in a later PR.

Reply:
OK. single quote will show $tplg_file not the real name in the log. It's OK to clean it up later

Reply:
@libinyang I misunderstand your point, now they are all fixed, and I check dlog* calls, all the arguments to dlog* function are quoted by `""`. Thanks for your reviewing.